malariagen_data/anoph/hapclust.py

```diff
@@ -0,0 +1,271 @@
+from typing import Optional, Tuple
+
+import allel  # type: ignore
+import numpy as np
+import pandas as pd
+from numpydoc_decorator import doc  # type: ignore
+
+from ..util import CacheMiss, check_types, pdist_abs_hamming
+from ..plotly_dendrogram import plot_dendrogram
+from . import base_params, plotly_params, tree_params, hap_params, hapclust_params
+from .base_params import DEFAULT
+from .snp_data import AnophelesSnpData
+from .hap_data import AnophelesHapData
+
+
+class AnophelesHapClust(AnophelesHapData, AnophelesSnpData):
```

Comment:
```suggestion
class AnophelesHapClustAnalysis(AnophelesHapData, AnophelesSnpData):
```

Nit, could we add "Analysis" to the class name, just for naming consistency.